### DISCUSSION TITLE
Use 'zero' instead of 'null' & 'default', or allow 'null' to be used with value types

### DISCUSSION THREAD

**Discussion by YT-GameMaker** [dotnet/csharplang#7655] [Language Ideas] [answered]
Introduce a new keyword, `zero`, representing the default state of any type. It can be used to replace `null` and `default`, unifying the syntax for reference types and value types, and providing a clearer distinction between the concepts of "uninitialized" and "new()" for value types.

If `zero` would still introduce ambiguity or if there's a desire to avoid adding new keywords, then `null` can also be made applicable to value types. In many cases, `null` can better represent that data is invalid compared to `0` or `default`.

**Comment by huoyaoyuan** [comment, marked as answer]
`default` is allowed for reference types. It already has the proposed semantics.

**Reply by tannergooding** [reply]
In addition to this, `null` has existing meaning and semantics in places like `generic` contexts and could not be changed to mean this without it being breaking, nor would it be desirable because `null` is explicitly a concept related to pointers/references/nullable value types. It is not applicable to general value types.

`default` is the existing keyword, representing the default state of any type. Hence the name: `default`

**Reply by YT-GameMaker** [reply]
`default` could be misleading as "new()", especially now that structs can have user-defined parameterless constructors, `default` and "new()" might be different, developers might not notice this aspect.

**Reply by CyrusNajmabadi** [reply]
The same holds for 'zero'. Ultimately, developers have to learn how the language works. We don't need to introduce synonyms to avoid non-existent problems.

**Comment by CyrusNajmabadi** [comment]
Closing out. We already have a literal and an expression for this (`default` and `default(T)`).
